airflow/cli/commands/dag_command.py

```diff
@@ -453,7 +453,10 @@ def dag_test(args, session=None):
     """Execute one single DagRun for a given DAG and execution date, using the DebugExecutor."""
     run_conf = None
     if args.conf:
-        run_conf = json.loads(args.conf)
+        try:
+            run_conf = json.loads(args.conf)
+        except ValueError as e:
+            raise SystemExit(f"Configuration '{args.conf}' is not valid JSON. Error: {e}")
```

Comment:
```suggestion
            raise SystemExit(f"Configuration {args.conf!r} is not valid JSON. Error: {e}")
```

This provides better escaping.